pasha/functor.py

```diff
@@ -267,15 +267,23 @@ def wrap(cls, value):
             return cls(value)
 
     def split(self, num_workers):
+        from os import getpid
+        self._parent_pid = getpid()
+
         return gen_split_slices(self.n_trains, n_parts=num_workers)
 
     def iterate(self, share):
         subobj = self.obj.select_trains(np.s_[share])
 
         # Close all file handles inherited from the parent collection
-        # to force re-opening them in each worker process.
-        for f in subobj.files:
-            f.close()
+        # to force re-opening them in each worker process, if they
+        # are different to the parent process that ran .split().
+        # For ProcessContext, `split` runs after forking and thus the
+        # property should not even be available.
```

Comment:
```suggestion
        # For ProcessContext, `split` runs in the parent after forking
        # and thus the attribute is not even available in the child.
```

The wording confused me a bit, because somehow I interpreted 'after forking' as 'in the child'. That might just be me, but maybe a few extra words don't hurt.

Reply:
Yes, comments such as this reveal that I wrote `pasha` almost exclusively with `multiprocessing` in mind. Thanks for catching it!

I have not made it explicit enough in the text, this PR is only meant to have a hotfix available yesterday. I daresay there should be another method in the `Functor` interface that allows for initialization before running in _childs_ (whatever that may mean depending on the `Context`). It does not feel very intuitive to (mis)use `Functor.spit` for this.